File: server/router/mine/mine_archives.go
package mine

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MineArchivesRouter struct{}

func (s *MineArchivesRouter) InitMineArchivesRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mineArchivesRouter := Router.Group("mineArchives").Use(middleware.OperationRecord())
	mineArchivesRouterWithoutRecord := Router.Group("mineArchives")
	mineArchivesRouterWithoutAuth := PublicRouter.Group("mineArchives")
	{
		mineArchivesRouter.POST("createMineArchives", mineArchivesApi.CreateMineArchives)
		mineArchivesRouter.DELETE("deleteMineArchives", mineArchivesApi.DeleteMineArchives)
		mineArchivesRouter.DELETE("deleteMineArchivesByIds", mineArchivesApi.DeleteMineArchivesByIds)
		mineArchivesRouter.PUT("updateMineArchives", mineArchivesApi.UpdateMineArchives)
	}
	{
		mineArchivesRouterWithoutRecord.GET("findMineArchives", mineArchivesApi.FindMineArchives)
		mineArchivesRouterWithoutRecord.GET("getMineArchivesList", mineArchivesApi.GetMineArchivesList)
	}
	{
		_ = mineArchivesRouterWithoutAuth
	}
}

File: web/src/api/mine/mine_archives.js
import service from '@/utils/request'

export const createMineArchives = (data) => {
  return service({
    url: '/mineArchives/createMineArchives',
    method: 'post',
    data
  })
}

export const deleteMineArchives = (params) => {
  return service({
    url: '/mineArchives/deleteMineArchives',
    method: 'delete',
    params
  })
}

export const deleteMineArchivesByIds = (params) => {
  return service({
    url: '/mineArchives/deleteMineArchivesByIds',
    method: 'delete',
    params
  })
}

export const updateMineArchives = (data) => {
  return service({
    url: '/mineArchives/updateMineArchives',
    method: 'put',
    data
  })
}

export const findMineArchives = (params) => {
  return service({
    url: '/mineArchives/findMineArchives',
    method: 'get',
    params
  })
}

export const getMineArchivesList = (params) => {
  return service({
    url: '/mineArchives/getMineArchivesList',
    method: 'get',
    params
  })
}
